docker/image/entrypoint.sh

```diff
@@ -1,21 +1,57 @@
 #!/usr/bin/env bash
 
-# Make sure all args given
+set -e
+
+echo "$(ls -nd /home/user | cut -f3 -d' ')"
+
 if [ -z ${VENV_PATH} ]; then
-    echo "Variable ENV_PATH for python virtual env is not defined!"
+    echo "Variable ENV_PATH for python virtual env is not defined! It should be set in docker image"
     return 1
 fi
 
-# Add here any commands that are required before starting python app
+# get uid/gid of repo owner (external user)
+EXTERNAL_UID=$(ls -nd /code | cut -f3 -d' ')
+EXTERNAL_GID=$(ls -nd /code | cut -f4 -d' ')
+
+if [ ! -z "$EXTERNAL_UID" -a ! -z "$EXTERNAL_GID" ]; then
+    echo "You user uid:gid is ${EXTERNAL_UID}:${EXTERNAL_GID}"
+
+    # get the uid/gid of docker "user"
+    INTERNAL_UID=$(getent passwd user | cut -f3 -d: || true)
+    INTERNAL_GID=$(getent group user | cut -f3 -d: || true)
+    echo "Internal user uid:gid is ${INTERNAL_UID}:${INTERNAL_GID}"
+
+    if [ "$EXTERNAL_GID" != "$INTERNAL_GID" ] || [ "$EXTERNAL_UID" != "$INTERNAL_UID"]; then
+        if [ "$(ls -nd /home/user | cut -f3 -d' ')" != "$EXTERNAL_UID" ]; then
+            echo "Updating files ownership from ${INTERNAL_UID} to ${EXTERNAL_UID}"
+            find / -mount -uid ${INTERNAL_UID} -exec chown ${EXTERNAL_UID}:${EXTERNAL_GID} {} \;
+        fi
+
+        echo "Updating internal uid:gid to ${EXTERNAL_UID}:${EXTERNAL_GID}"
+        usermod -u ${EXTERNAL_UID} user
+        groupmod -g ${EXTERNAL_GID} user
```

Comment:
It's hard to me to judge at first glance how crazy this is but Iet's see it at work :)

Reply:
I can only assure you that it is not crazy at all and hope future will prove this ;)